README.md

```diff
@@ -172,245 +197,193 @@ Ipld::DagJson::serialize(
+Data already parsed to an in-memory IPLD representation can be canonically encoded trivially: it has already been through a [parser / validator][Parse Don't Validate].
 
-Data purporting to conform to an IPLD encoding (such as [DAG-JSON]) MUST be validated prior to signature verification. This MAY be as simple as round-trip decoding/encoding the JSON and checking that the hash matches. A validation error MUST be signalled if it does not match.
+Data purporting to conform to an IPLD encoding (such as [DAG-JSON]) MUST be validated prior to signature verification. This MAY be as simple as round-trip decoding/encoding the JSON and checking that the hash matches. A validation error MUST be signaled if it does not match.
 
-> [Implementers] may provide an opt-in for systems where round-trip determinism is a desireable [sic] feature and backward compatibility with old, non-strict data is unnecessary.
+> Implementers may provide an opt-in for systems where round-trip determinism is a desireable [sic] feature and backward compatibility with old, non-strict data is unnecessary.
 >
-> — [DAG-JSON Spec][DAG-JSON]
+> — [DAG-JSON Spec][DAG-JSON]
 
 As it is critical for guarding against various attacks, the assumptions around canonical encoding MUST be enforced.
 
-## 2.1 Signing CIDs
+## Signing CIDs
 
-Rather than validating the inline IPLD, replacing the data with a CID link to the content MAY be used instead. Note while this is very safe (as it is impractical to alter a signed hash), this approach mixes data layout with security, and may have a performance, disk, and networking impacts.
+Rather than validating the inline IPLD, replacing the data with a CID link to the content MAY be used instead. Note while this is very safe (as it is impractical to alter a signed hash), this approach mixes data layout with security, and may have a performance, storage, data availability, and networking consequences.
 
-### 2.1.1 Caching & Invalidation
+### Caching & Invalidation
 
 Signing CIDs has two additional caching consequences:
 
 1. Signing CIDs enables a simple strategy for caching validation by CID.
 2. Such a strategy also MAY require accounting for revocation of the signing keys themselves. In this case, the cache would need to include additional information about the signing key.
 
-## 2.2 Raw (Noncanonicalized) Data
+# Format
 
-Canonicalization is not required if data is encoded as raw bytes (multicodec `%x55`). The exact bytes are already present, and MUST not be changed.
+A varsig has two main segments:
 
-# 3 Varsig Format
+* [Header]: binary coded signature metadata
+* [Signature]: the signature itself
 
-After being decoded from [unsigned varint]s, a varsig includes the following segments:
+Including the varsig header in the payload that is signed over is RECOMMENDED. Doing so eliminates any ambiguity of the signed payload format and signature algorithm configuration.
```

Comment:
either we make it REQUIRED to include the varsig header in the signature or we need to signal inclusion in the header so its clear how to verify

Reply:
Added a section per our out of band conversation 👍